src/client.rs

```diff
@@ -0,0 +1,850 @@
+			Ok(r) => Ok(r),
+			Err(e) => Err(e)
+    }
+	}
+
+	/// Returns the height of most recent block.
+	///
+	/// # Arguments
+	///
+	/// * `none`
+	/// 
+	/// # Returns
+	/// 
+	/// The current block height the client is on.
+	///
+	/// # Example
+	///
+	/// ```
+	/// use nimiq_rpc::Client;
+	/// let client = Client::new("http://seed-host.com:8648");
+	/// let result = client.block_number();
+	/// ```
+	pub fn block_number(&self) -> Result<u64, Error>{
+		let request = self.agent.build_request("blockNumber", &[]);
+    match self.agent.send_request(&request).and_then(|res| res.into_result::<u64>()) {
+			Ok(r) => Ok(r),
+			Err(e) => Err(e)
+    }
+	}
+
+	/// Returns information on the current consensus state.
+	///
+	/// # Arguments
+	///
+	/// * `none`
+	/// 
+	/// # Returns
+	/// 
+	/// String describing the consensus state. `"established"` is the value for a good state, other values indicate bad state.
+	///
+	/// # Example
+	///
+	/// ```
+	/// use nimiq_rpc::Client;
+	/// let client = Client::new("http://seed-host.com:8648");
+	/// let result = client.consensus();
+	/// ```
+	pub fn consensus(&self) -> Result<String, Error>{
+		let request = self.agent.build_request("consensus", &[]);
+    match self.agent.send_request(&request).and_then(|res| res.into_result::<String>()) {
+			Ok(r) => Ok(r),
+			Err(e) => Err(e)
+    }
+	}
+
+	/// Creates a new account and stores its private key in the client store.
+	///
+	/// # Arguments
+	///
+	/// * `none`
+	/// 
+	/// # Returns
+	/// 
+	/// Information on the wallet that was created using the command.
+	///
+	/// # Example
+	///
+	/// ```
+	/// use nimiq_rpc::Client;
+	/// let client = Client::new("http://seed-host.com:8648");
+	/// let result = client.create_account();
+	/// ```
+	pub fn create_account(&self) -> Result<primitives::Wallet, Error>{
+		let request = self.agent.build_request("createAccount", &[]);
+    match self.agent.send_request(&request).and_then(|res| res.into_result::<primitives::Wallet>()) {
+			Ok(r) => Ok(r),
+			Err(e) => Err(e)
+    }
+	}
+
+	/// Creates and signs a transaction without sending it. The transaction can then be send via `sendRawTransaction` without accidentally replaying it.
+	///
+	/// # Arguments
+	///
+	/// * `raw_transaction`: `&primitives::OutgoingTransaction`
+	/// 
+	/// # Returns
+	/// 
+	/// The Hex-encoded transaction.
+	///
+	/// # Example
+	///
+	/// ```
+	/// use nimiq_rpc::Client;
+	/// let client = Client::new("http://seed-host.com:8648");
+	/// let tx = nimiq_rpc::primitives::OutgoingTransaction {
+	///    from: "NQ32 R6DB VFM5 M931 7X4E 0N5Q LJ56 9QCR 4T42",
+	///    to: "NQ74 61S8 2FD3 RVPG HU09 1Y57 77E6 BL38 TQH3",
+	///    value: 100, //Lunas
+	///    fee: 0
+	///	};
+	/// let result = client.create_raw_transaction(&tx);
+	/// ```
+	pub fn create_raw_transaction(&self, raw_transaction: &primitives::OutgoingTransaction) -> Result<String, Error>{
+		let params = &[
+			serde_json::to_value(&raw_transaction).unwrap()
```

Comment:
You shouldn't call `unwrap` here. I guess the idea was that it won't really be able to fail.
But I think it would be better to pass the error outside of the function.
```suggestion
			serde_json::to_value(&raw_transaction)?
```
Should probably do the job.